### DISCUSSION TITLE
[5.5.x] Link field data in GraphQL

### DISCUSSION THREAD

**Discussion by yingban** [craftcms/cms#16104] [Ideas] [answered]
Hi,

I just tested the link field label and target with GraphQL, but it returns a string.
Would it be possible to return an object including all data?

- label
- type 
- url
- target
- etc.

**Comment by AmsalSugihan** [comment]
Got the same issue here :D

**Comment by franck-thorck** [comment]
Same here! I temporally use Hyper plugin (or we can create new entry type as a link field) 🙂

**Comment by brandonkelly** [comment, marked as answer]
Just added a new “GraphQL Mode” setting to Link fields for Craft 5.6, which determines how Link fields should interface with GraphQL going forward. (#16237)

**Reply by brandonkelly** [reply]
Craft 5.6.0 is out now 🚀

**Comment by diluno** [comment]
Would it be possible to add the entry URI to the GraphQL data? It would make it a lot easier to create relative links in SPAs.

**Reply by brandonkelly** [reply]
Sure, just added `elementSlug` and `elementUri` for Craft 5.7 (8e12fc0c6b17662237a9bdf3f80c7727054e5387).

**Reply by brandonkelly** [reply]
Edit: I removed `elementSlug` and `elementUri` in favor of `element`, `asset`, `entry`, etc. – which are references to the linked element, so you can fetch whatever properties you want off of those. (#16698)

**Reply by diluno** [reply]
Perfect, thank you so much!! Have a nice weekend :)

**Reply by brandonkelly** [reply]
Craft 5.7.0 is out with that change.
